src/randoop/operation/ConstructorCall.java

```diff
@@ -149,32 +146,39 @@ public void appendCode(Variable varName, List<Variable> inputVars, StringBuilder
       // or string literal, like "int x = 3" are not added to a sequence;
       // instead, the value (e.g. "3") is inserted directly added as
       // arguments to method calls.
-      Operation statementCreatingVar = inputVars.get(i).getDeclaringStatement(); 
-      if (!GenInputsAbstract.long_format
-          && ExecutableSequence.canUseShortFormat(statementCreatingVar)) {
-        b.append(PrimitiveTypes.toCodeString(((NonreceiverTerm) statementCreatingVar).getValue()));
+      Statement statementCreatingVar = inputVars.get(i).getDeclaringStatement(); 
+      String shortForm = statementCreatingVar.getShortForm();
+      if (!GenInputsAbstract.long_format && shortForm != null) {
+        b.append(shortForm);
       } else {
         b.append(inputVars.get(i).getName());
       }
     }
-    b.append(");");
-    b.append(Globals.lineSep);
+    b.append(")");
+    
   }
 
+  /**
+   * Tests whether the parameter is a call to the same constructor.
+   * 
+   * @param o  an object
+   * @return true if o is a ConstructorCall referring to same constructor object; false otherwise.
+   */
   @Override
   public boolean equals(Object o) {
-    if (o == null)
-      return false;
-    if (this == o)
-      return true;
-    if (!(o instanceof ConstructorCall))
-      return false;
-    ConstructorCall other = (ConstructorCall) o;
-    if (!this.constructor.equals(other.constructor))
-      return false;
-    return true;
+    if (o instanceof ConstructorCall) {
```

Comment:
Same comment as before about canonical way to write an equals() method implementation.